src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -303,6 +304,29 @@ public Blob createGcsFile(BlobId blobId, String projectId) {
    */
   public void validateUserCanRead(
       List<String> sourcePaths, String cloudEncapsulationId, AuthenticatedUserRequest user) {
+    validateUserCanRead(sourcePaths, cloudEncapsulationId, user, true);
+  }
+
+  /**
+   * Given a list a source paths, validate that the specified user has a pat service account with
+   * read permissions
+   *
+   * @param sourcePaths A list of gs:// formatted paths
+   * @param cloudEncapsulationId The dataset project to bill to if any of the source buckets are
+   *     configured to use requester pays
+   * @param user An authenticated user
+   * @param addPSAToDatasetProject if true, grant permissions on the dataset's project to the user's
+   *     pet service account so that it can be used to ingest data. Note: this should only be true
+   *     when ingesting into a GCP backed dataset. In the case of Azure backed datasets, there is no
+   *     such bucket and the ingest mechanism does not require it
+   * @throws BlobAccessNotAuthorizedException if the user does not have an authorized pet
+   * @throws IllegalArgumentException if the source path is not a valid blob url
+   */
+  public void validateUserCanRead(
+      List<String> sourcePaths,
+      String cloudEncapsulationId,
+      AuthenticatedUserRequest user,
+      boolean addPSAToDatasetProject) {
```

Comment:
As a heads up, I'm also modifying this method as part of skipping permission validation for dedicated SA datasets!

I bring it up because I ended up passing the whole dataset to `validateUserCanRead`, and adding a helper method `Dataset.shouldValidateIngesterFileAccess()`.  That way the logic stays close to the dataset itself.  I can throw up a draft PR if it helps.

What do you think about this approach instead, especially as we'll be supporting multiple conditional behavior mods?

Reply:
Awesome! Lights a fire under me to get this out, ha! (I don't mind rebasing off of yours though.)

Reply:
haha, no fire needed

Reply:
As discussed in stand-up, this PR is likely to go in before [mine](https://github.com/DataBiosphere/jade-data-repo/pull/1453). I will rebase and pull the dataset through :)